www/js/services/shareLocalDBFile.ts

```diff
@@ -0,0 +1,121 @@
+        parentDirectory = window['cordova'].file.dataDirectory.replace('files', 'databases');
+        // pathToFile: /data/user/0/edu.berkeley.eecs.emission/files/
+        pathToFile = parentDirectory.replace('file://', '') + fileName;
+      } else if (window['cordova'].platformId == 'ios') {
+        // parentDirectory: file:///var/mobile/Containers/Data/Application/<32-hex-digit-id>/Library/NoCloud/../
+        parentDirectory = window['cordova'].file.dataDirectory + '../';
+        pathToFile = 'LocalDatabase/' + fileName;
+      } else {
+        displayErrorMsg('Error: Unknown OS!');
+        throw new Error('Error: Unknown OS!');
+      }
+
+      window['resolveLocalFileSystemURL'](parentDirectory, (fs) => {
+        // On iOS, pass in relative path to getFile https://github.com/e-mission/e-mission-phone/pull/1160#issuecomment-2192112472
+        // On Android, pass in absolute path to getFile https://github.com/e-mission/e-mission-phone/pull/1160#issuecomment-2204297874
+        fs.filesystem.root.getFile(pathToFile, { create: false, exclusive: false }, (fileEntry) => {
+          // logDebug(`fileEntry ${fileEntry.nativeURL} is file? ${fileEntry.isFile.toString()}`);
+          logDebug(`fileEntry is: ${JSON.stringify(fileEntry, null, 2)}`);
+          window['resolveLocalFileSystemURL'](window['cordova'].file.cacheDirectory, (copyDir) => {
+            logDebug(`DirectoryEntry is: ${JSON.stringify(copyDir.filesystem.root, null, 2)}`);
+
+            fileEntry.copyTo(
+              copyDir.filesystem.root,
+              fileName + fileExtension,
+              (res) => {
+                logDebug(`Res: ${res}`);
+                resolve();
+              },
+              (rej) => {
+                displayErrorMsg(`Rej: ${JSON.stringify(rej, null, 2)}`);
+                reject();
+              },
+            );
+          });
+        });
+      });
+    });
+  }
+
+  function localShareFile() {
+    return new Promise<void>((resolve, reject) => {
+      window['resolveLocalFileSystemURL'](window['cordova'].file.cacheDirectory, (fs) => {
+        fs.filesystem.root.getFile(
+          fileName + fileExtension,
+          null,
+          (fileEntry) => {
+            const shareObj = {
+              files: [fileEntry.nativeURL],
+              message: i18next.t('shareFile-service.send-log.body-please-fill-in-what-is-wrong'),
+              subject: i18next.t('shareFile-service.send-log.subject-logs'),
+            };
+            window['plugins'].socialsharing.shareWithOptions(
+              shareObj,
+              (result) => {
+                logDebug(`Share Completed? ${result.completed}`); // On Android, most likely returns false
+                logDebug(`Shared to app:  ${result.app}`);
+                resolve();
+              },
+              (msg) => {
+                displayErrorMsg(`Sharing failed with message ${msg}`);
+              },
```

Comment:
@the-bay-kay @louisg1337 this should give you much better formatting

```suggestion
              (error) => {
                displayError(error, `Sharing failed with error`);
              },
```